docs/README.md

```diff
@@ -0,0 +1,256 @@
+
+```ruby
+require 'uri'
+require 'cgi'
+require 'date'
+require 'json'
+require 'base64'
+require 'erb'
+require 'securerandom'
+require 'time'
+require 'timeliness'
+require 'faraday'
+require 'faraday-cookie_jar'
+require 'concurrent'
+require 'ms_rest'
+require 'ms_rest_azure'
+
+options = {
+  tenant_id: 'YOUR TENANT ID',
+  client_id: 'YOUR CLIENT ID',
+  client_secret: 'YOUR CLIENT SECRET',
+  subscription_id: 'YOUR SUBSCRIPTION ID'
+}
+
+# Create a client
+provider = MsRestAzure::ApplicationTokenProvider.new(
+       options[:tenant_id],
+       options[:client_id],
+       options[:client_secret])
+credentials = MsRest::TokenCredentials.new(provider)
+client = MsRestAzure::AzureServiceClient.new(credentials, options)
+
+# Set Headers
+request_headers = {
+    'Content-Type': 'application/json; charset=utf-8',
+    'x-ms-client-request-id': SecureRandom.uuid,
+    'accept-language': 'en-US'
+}
+
+request_url = 'https://management.azure.com'
+path_template = 'subscriptions/{subscriptionId}/providers/Microsoft.Storage/checkNameAvailability'
+api_version = '2019-06-01'
+
+# Set Content
+request_content = {
+    name: "sto3363",
+    type: "Microsoft.Storage/storageAccounts"
+}
+request_content = JSON.generate(request_content, quirks_mode: true)
+
+# Set operations
+request_options = {
+    middlewares: [
+        [MsRest::RetryPolicyMiddleware, times: 3, retry: 0.02], # Add retry policy middleware
+        [:cookie_jar] # Add faraday cookie jar middleware
+    ],
+    path_params: {
+        'subscriptionId': options[:subscription_id],
+    },
+    query_params: {
+        'api-version': api_version
+    },
+    body: request_content,
+    headers: request_headers,
+    base_url: request_url
+}
+
+# Make request
+promise = client.make_request_async(request_url, :post, path_template, request_options)
+
+promise = promise.then do |result|
+  http_response = result.response
+  status_code = http_response.status
+  response_content = http_response.body
+
+  unless status_code == 200
+    error_model = JSON.load(response_content)
+    fail MsRestAzure::AzureOperationError.new(result.request, http_response, error_model)
+  end
+
+  result.request_id = http_response['x-ms-request-id'] unless http_response['x-ms-request-id'].nil?
+  result.correlation_request_id = http_response['x-ms-correlation-request-id'] unless http_response['x-ms-correlation-request-id'].nil?
+  result.client_request_id = http_response['x-ms-client-request-id'] unless http_response['x-ms-client-request-id'].nil?
+  
+  # Deserialize Response
+  if status_code == 200
+    begin
+      result.body = response_content.to_s.empty? ? nil : JSON.load(response_content)
+    rescue Exception => e
+      fail MsRest::DeserializationError.new('Error occurred in deserializing the response', e.message, e.backtrace, result)
+    end
+  end
+
+  result
+end
+
+response = promise.execute.value!
+
+```
+
+# Provide feedback
```

Comment:
refine this section to be more like https://github.com/Azure/azure-sdk-for-java/#need-help